reqs_dir.go

```diff
@@ -95,7 +95,11 @@ const readdirResponseHeaderStart = unsafe.Offsetof(ReaddirResponse{}.outHeader)
 const readdirResponseSize = unsafe.Sizeof(ReaddirResponse{})
 
 func readdirResponse(a *Allocator, dataSize uint32) *ReaddirResponse {
-	return (*ReaddirResponse)(a.allocPointer(readdirResponseBaseSize + uintptr(dataSize)))
+	bytes := a.alloc(readdirResponseBaseSize+uintptr(dataSize), false)
```

Comment:
I think I don't grok why we would or wouldn't want to zero out memory. Why would we ever want to not zero it out? (the answer should probably be in the code)